lib/matplotlib/patches.py

```diff
@@ -1552,9 +1552,189 @@ def get_angle(self):
     angle = property(get_angle, set_angle)
 
 
-class Circle(Ellipse):
-    """A circle patch."""
+class Annulus(Patch):
+    """
+    An elliptical annulus.
+    """
+
+    @docstring.dedent_interpd
+    def __init__(self, xy, r, width, angle=0.0, **kwargs):
+        """
+        xy : (float, float)
+            xy coordinates of annulus centre.
+        r : float or (float, float)
+            The radius, or semi-major axes
+                - If float: radius of the outer circle
+                - If two floats: semi-major and -minor axes of outer
+                  ellipse
+        width : float
+            Width (thickness) of the annulus ring.
```

Comment:
How exactly do r and width relate to the annulus? In which direction from the r ellipse does width go, in, out or half in both directions?

Also, are we sure that `r, width` is the way we want to specify this? An alternative would be the inner and outer ellipse via `r1, r2` (but that'd be a bit more general. You could get non-constant width, which is ok, but you could also make the two ellipses intersect, which may or may not give reasonable rendering.)

Reply:
@timhoffm `r` is the radius of the outer ellipse. `width` is measured inward from the outer radius, so the radius of the inner ellipse is `r - width`

```python
fig, ax = plt.subplots(subplot_kw=dict(aspect='equal'))
ax.add_patch(Annulus((0, 0), r=1, width=0.4, fc='g') )
```
![ann](https://user-images.githubusercontent.com/5623552/113835281-1f395a00-978c-11eb-81e8-8876b049b6b3.png)

I found using `width` as an initializer to be the most natural option, but I suppose this is subjective. Afaict, it is just as general as specifying the two radii. Since the inner ellipse is just a rescaled version of the outer, at the same tilt angle, they will never intersect. `width` might be somewhat confusing though to users that are used to `Ellipse` which uses `width` and `height` as its main parameters (meaning 2x semi-major and semi-minor axes). 

Happy to re-parametrize the initializer to use the two radii if that seems best. Another option might be a alternative `classmethod` as constructor using the radii: Something like:
```python
Annulus.from_radii(xy, r0, r1, angle, **kws)
```

Reply:
Update: I think I see what you mean regarding greater generality of using `r0, r1` and non-constant width in the case that both are array-like of size 2. Implementing this would require a rewrite of the `_recompute_path` method.  I'm not convinced that it would really be worth it - i suspect the fully general case will not see much use.

Reply:
Ok, let's keep it with one width for now. I wanted to make sure that this is a conscious decision - and that we're not accidentally ending up with a too specific API.

On a side-note: We can still generalize using two widths later if the need arises.

Please clarify the `width` docs concerning the "pointing inward" semantic.